### PR TITLE
[LTD-4230] Case sub status ordering

### PR DESCRIPTION
**Pull request by currycoder**
### Aim

Ensure that the list case sub status endpoint returns sub statuses in the correct order for the application.  This is achieved by adding a new `order` field so that we are able to record ordering in the DB; the field is used for sorting in the queryset for the view.

[LTD-4230](https://uktrade.atlassian.net/browse/LTD-4230)

[LTD-4230]: https://uktrade.atlassian.net/browse/LTD-4230?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/staticdata/statuses/models.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ class CaseSubStatus(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     name = models.CharField(max_length=50, unique=True)
     parent_status = models.ForeignKey(CaseStatus, on_delete=models.CASCADE, related_name="sub_statuses")
+    order = models.DecimalField(null=False, blank=False, max_digits=5, decimal_places=2, default=10)
```
[the remark this answers is not in the document]
Comment (currycoder) [reply api/staticdata/statuses/models.py:33]: Maybe this is early optimisation, but something that has made adjusting ordering fields annoying for me in the past is having continuous runs of integers.  So I felt like a decimal field would give us a bit more flexibility without having to think about it too hard.  e.g. being able to insert a 1.5 between records with `order` 1 and 2 etc. What do you think?

[the remark this answers is not in the document]
Comment (kevincarrogan) [reply api/staticdata/statuses/models.py:33]: I would have done the old school BASIC trick and just made the initial order separated by multiples of 10 i.e. 10, 20, 30 etc.

[the remark this answers is not in the document]
Comment (currycoder) [reply api/staticdata/statuses/models.py]: Yup makes sense, I've flipped it to that way of doing things.  Will squash commits when this is approved

[the remark this answers is not in the document]
Comment (saruniitr) [reply api/staticdata/statuses/models.py]: multiples of 10 is a neat idea.
@kevincarrogan the reference to BASIC, do you mean the line numbers we write on the left or any other trick?

**Review by saruniitr — APPROVED** [review]

File: api/staticdata/statuses/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply api/staticdata/statuses/models.py]: Yeah. The BASIC trick is from when you had to write line numbers on the left so you could do `GOTO 10`, the multiples of 10 is for this same reason as this where you might later want to put a new line in between some others without having to change all of the line numbers.

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**